document-store/src/main/java/org/hypertrace/core/documentstore/UpdateResult.java

```diff
@@ -0,0 +1,16 @@
+package org.hypertrace.core.documentstore;
+
+/*
+ * Represent the result object for UPDATE operation of document store APIs.
+ * */
+public class UpdateResult {
+  private long updatedCount;
+
+  public UpdateResult(long updatedCount) {
+    this.updatedCount = updatedCount;
+  }
+
+  public long getUpdatedCount() {
+    return updatedCount;
```

Comment:
Is this absolutely necessary? Also, can't we make it lazy? In general I think this result should be an interface as some databases retrieve this values for free and others have to perform operations so I'd leave this up to the implementation.

Reply:
> Also, can't we make it lazy?
Didn't get it.